### PR TITLE
Feature robos core

### PR DESCRIPTION
**Pull request by GregoryRochat**
Pulling all the latest added functionality of module update and run functions from robos core to release.

### CODE REVIEW

File: code/src/robos_core.cpp [hunk 1]
```diff
@@ -77,20 +77,45 @@ namespace r2d2::robos {
...
+            // auto moving_platform_role = robos_role_c();
+            // robos_core_c::robos role = moving_platform;
             break;
         }
+
         return 0;
     };
 
     int robos_core_c::run_role() {
+        const std::vector<frame_type> test_frame;
```
Comment (MarkGasse) [inline code/src/robos_core.cpp]: vector should be an array

**Review by MarkGasse — CHANGES_REQUESTED** [review]

File: code/src/robos_core.cpp (hunk 1 above)
Comment (olivierverwoerd) [reply code/src/robos_core.cpp]: This is an exception made by the leads that in this case a vector is correct. So i will approve

**Review by olivierverwoerd — APPROVED** [review]

File: code/src/robos_core.cpp (hunk 1 above)
Comment (MarkGasse) [reply code/src/robos_core.cpp]: Ok

File: code/src/robos_core.cpp [hunk 2]
```diff
@@ -77,20 +77,45 @@ namespace r2d2::robos {
...
+        robos_core_c::current_role->run(test_frame);
```
Comment (olivierverwoerd) [inline code/src/robos_core.cpp]: robos_core_c::current_role->run(test_frame); causes error
robos_core.cpp:(.text._ZN4r2d25robos12robos_core_c8run_roleEv+0x1a): undefined reference to `operator delete(void*)'
collect2.exe: error: ld returned 1 exit status

**Review by olivierverwoerd — CHANGES_REQUESTED** [review]

**Review by olivierverwoerd — APPROVED** [review]
Dont think out commenting lines of code is a good solution but if it helps the development now i approve

**Review by MarkGasse — APPROVED** [review]

**Pull request merged**